### PR TITLE
support np.int16, np.uint16 and np.float16

### PR DESCRIPTION
**Pull request by ledmonster**
I added `np.int16`, `np.uint16` and `np.float16` support. I'm not familiar with rust, so I appreciate your feedback. Thanks.

### CODE REVIEW

File: test/test_numpy.py [hunk 1]
```diff
@@ -371,10 +416,23 @@ def test_numpy_scalar_int8(self):
             b"127",
         )
         self.assertEqual(
-            orjson.dumps(numpy.int8(--128), option=orjson.OPT_SERIALIZE_NUMPY),
```
Comment (ledmonster) [inline test/test_numpy.py:374]: This test seems to be wrong. I'm not sure why CI didn't blame it.

**Comment by ijl** [comment]
CI doesn't run auomatically. Please setup Rust locally and try to build. float16 is not in Rust and probably requires work to a lot of dependencies. i16 and u16 would be fine to add.

**Comment by ijl** [comment]
Feel free to come back to it.

**Comment by ledmonster** [comment]
Sorry for replying late. I see, thanks!

**Comment by NazarKostetskiy** [comment]
Is it possible to add only i16 & u16 from this PR and add float16 support later, as it's more complex? It would be nice to have it.

**Pull request closed without merging**